src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
+Animal.alive = [];
+
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivoreAnimal) {
+    const eatenAnimal = herbivoreAnimal;
+
+    if (!eatenAnimal.hidden && !(eatenAnimal instanceof Carnivore)) {
+      eatenAnimal.health -= 50;
+
+      if (eatenAnimal.health <= 0) {
```

Comment:
```suggestion
      if (eatenAnimal.health === 0) {
```
I don`t think it can be less then 0)